src/client/kube_store/index.ts

```diff
@@ -233,6 +47,48 @@ export default class KubeStore {
     return this.instance || (this.instance = new this(''));
   }
 
+  public patchResource(resource: KubeResource, name: string, patch: object): object {
+    let result: object = {};
+    let key: string = '';
+
+    if (resource instanceof NamespacedResource) {
+      const namespacedResource = resource as NamespacedResource;
+      if (resource instanceof CoreNamespacedResource) {
+        // Core Namespaced
+        key = `api/${resource.gvk().version}/namespaces/${namespacedResource.namespace}/${resource.gvk().kindPlural}`;
+        if (this.data()[key]) {
+          result = this.data()[key][name];
+        }
+      } else {
+        // Extension Namespaced
+        key = `apis/${resource.gvk().group}/${resource.gvk().version}` +
+          `/namespaces/${namespacedResource.namespace}/${resource.gvk().kindPlural}`;
+        if (this.data()[key]) {
+          result = this.data()[key][name];
+        }
+      }
+    } else {
+      if (resource instanceof CoreClusterResource) {
+        // Core ClusterResource
+        key = `api/${resource.gvk().version}/namespaces/${resource.gvk().kindPlural}`;
+        if (this.data()[key]) {
+          result = this.data()[key][name];
+        }
+      } else {
+        // Extension ClusterResource
+        key = `apis/${resource.gvk().group}/${resource.gvk().version}/${resource.gvk().kindPlural}`;
+        if (this.data()[key]) {
+          result = this.data()[key][name];
+        }
+      }
+    }
+
+    const patchedObject = JsonMergePatch.apply(resource, patch);
+    localStorage[key] = patchedObject;
```

Comment:
This is wrong.

As we discussed, we are storing the entire mocked data as a single key in localStorage.
So we have a single key for ALL of our data.

This method as written implies that we are storing individual elements in localStorage with a primary key, we are not doing that.  We are using a series of nested keys.

Reply:
If I follow this correctly, the localStorage should contain the entire mocked data + patch, so it'd be like 
```
    const localStorageData = {...mocked_data, ...patchedObject};
    localStorage.setItem(localStorageMockedDataKey, JSON.stringify(localStorageData))
```